drv/sprot-api/src/lib.rs

```diff
@@ -481,23 +489,27 @@ pub struct RxMsg {
     buf: [u8; BUF_SIZE],
 }
 
+impl AsMut<[u8]> for RxMsg {
+    fn as_mut(&mut self) -> &mut [u8] {
+        &mut self.buf[..]
+    }
+}
+
+impl Default for RxMsg {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl RxMsg {
     pub fn new() -> RxMsg {
         RxMsg { buf: [0; BUF_SIZE] }
     }
 
-    pub fn len(&self) -> usize {
-        self.buf.len()
-    }
-
```

Comment:
I don't think `len()` is provided by `AsMut<_>` is it?

Reply:
Nevermind, you fixed this already

Reply:
No, I'm removing this entirely to fix a warning about the lack of `is_empty`.

(you can still get the length with `as_slice().len()`, which is what we do in `TxBuf` already)